tangelo/tangelo/__main__.py

```diff
@@ -194,6 +189,19 @@ def main():
         print tangelo_version
         return 0
 
+    # Make all logging go to stdout instead of stderr.
+    #
+    # The "cherrypy.error" logger contains two handlers by default - a
+    # NullHandler, and a StreamHandler that writes to stderr.  This piece of
+    # code simply removes the StreamHandler, modifies it to write to stdout, and
+    # then replaces it.
+    import logging
+    logger = logging.getLogger("cherrypy.error")
+    handler = logger.handlers[1]
+    logger.removeHandler(handler)
+    handler.stream = sys.stdout
+    logger.addHandler(handler)
+
```

Comment:
Without this bit of code, the output gets split between stderr and stdout,
which is troublesome for anyone scraping the output, as I was doing when
writing the config tests (since their correctness depends on certain
logging output).  The same problem will hit anyone else looking at
Tangelo's output in a specific way, including general system services that
turn output streams into logs on disk, etc.

I would also be ok with making all output go to stderr instead, but Girder
sends some of its own output to stdout so I figured we could "standardize"
on that.

On Wed, Jan 7, 2015 at 7:50 AM, Jeffrey Baumes <notifications@github.com>
wrote:

> In tangelo/tangelo/__main__.py
> <https://github.com/Kitware/tangelo/pull/469#discussion-diff-22584224>:
>
> > @@ -194,6 +189,19 @@ def main():
> >          print tangelo_version
> >          return 0
> >
> > +    # Make all logging go to stdout instead of stderr.
> > +    #
> > +    # The "cherrypy.error" logger contains two handlers by default - a
> > +    # NullHandler, and a StreamHandler that writes to stderr.  This piece of
> > +    # code simply removes the StreamHandler, modifies it to write to stdout, and
> > +    # then replaces it.
> > +    import logging
> > +    logger = logging.getLogger("cherrypy.error")
> > +    handler = logger.handlers[1]
> > +    logger.removeHandler(handler)
> > +    handler.stream = sys.stdout
> > +    logger.addHandler(handler)
> > +
>
> For background: Why do we need to make everything go to stdout?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Kitware/tangelo/pull/469/files#r22584224>.
>